### PR TITLE
Configuring for pure-python

### PR DESCRIPTION
**Pull request by icemac**
For reference I added Python 2.7 tests which are the only ones which do not fail.

### CODE REVIEW

**Comment by ale-rt** [comment]
I think I have a valid patch for this but I cannot push it for some reason:
```patch
ale@avo:~/Code/plone/packages/zc.relation$ git diff origin/config-with-zope-product-template-096e23ce |cat
diff --git a/src/zc/relation/CHANGES.rst b/src/zc/relation/CHANGES.rst
index 00ea53a..b6203ae 100644
--- a/src/zc/relation/CHANGES.rst
+++ b/src/zc/relation/CHANGES.rst
@@ -6,7 +6,10 @@ Changes
 1.2 (unreleased)
 ================
 
-- Nothing changed yet.
+- Adapt code for PEP-479 (Change StopIteration handling inside generators).
+  See: https://peps.python.org/pep-0479.
+  Fixes #11.
+  [ale-rt]
 
 
 1.1.post2 (2018-06-18)
diff --git a/src/zc/relation/catalog.py b/src/zc/relation/catalog.py
index f78d306..7745ee7 100644
--- a/src/zc/relation/catalog.py
+++ b/src/zc/relation/catalog.py
@@ -40,6 +40,8 @@ RELATION = None
 # helpers
 #
 
+_marker = object()
+
 
 def multiunion(sets, data):
     sets = tuple(s for s in sets if s)  # bool is appropriate here
@@ -917,9 +919,8 @@ class Catalog(persistent.Persistent):
             stack.append(((), iter(d)))
         while stack:
             tokenChain, relDataIter = stack[0]
-            try:
-                relToken = next(relDataIter)
-            except StopIteration:
+            relToken = next(relDataIter, _marker)
+            if relToken is _marker:
                 stack.pop(0)
             else:
                 tokenChain += (relToken,)
@@ -1177,13 +1178,13 @@ class Catalog(persistent.Persistent):
         if getQueries is None:
             queryFactory, getQueries = self._getQueryFactory(
                 query, queryFactory)
-        try:
-            next(self.yieldRelationTokenChains(
+
+        _ = next(self.yieldRelationTokenChains(
                 *self._parse(
                     query, maxDepth, filter, targetQuery,
                     targetFilter, getQueries) +
-                (False,)))
-        except StopIteration:
+                (False,)), _marker)
+        if _ is _marker:
             return False
         else:
             return True
diff --git a/src/zc/relation/searchindex.py b/src/zc/relation/searchindex.py
index 3a741e0..4533705 100644
--- a/src/zc/relation/searchindex.py
+++ b/src/zc/relation/searchindex.py
@@ -437,19 +437,24 @@ class Intransitive(persistent.Persistent):
         for name in self.names:
             src = source[name]
             iterator = iter(src)
-            value = next(iterator)  # should always have at least one
+            value = next(iterator, _marker)  # should always have at least one
+            if value is _marker:
+                return
             vals.append([name, value, iterator, src])
         while 1:
             yield BTrees.family32.OO.Bucket(
                 [(name, value) for name, value, iterator, src in vals])
             for s in vals:
                 name, value, iterator, src = s
-                try:
-                    s[1] = next(iterator)
-                except StopIteration:
+                _ = next(iterator, _marker)
+                if _ is _marker:
                     iterator = s[2] = iter(src)
-                    s[1] = next(iterator)
+                    _ = next(iterator, _marker)
+                    if _ is _marker:
+                        return
+                    s[1] = _
                 else:
+                    s[1] = _
                     break
             else:
                 break
```

@icemac, @dataflake can you push a new branch to this repo?
```
[ale@flo tmp]$ git clone git@github.com:zopefoundation/zc.relation.git
Cloning into 'zc.relation'...
remote: Enumerating objects: 558, done.
remote: Counting objects: 100% (57/57), done.
remote: Compressing objects: 100% (38/38), done.
remote: Total 558 (delta 23), reused 35 (delta 10), pack-reused 501
Receiving objects: 100% (558/558), 399.66 KiB | 1.15 MiB/s, done.
Resolving deltas: 100% (279/279), done.
🌠 ~/tmp  
[ale@flo tmp]$ cd zc.relation/
🌠 ~/tmp/zc.relation (master) 
[ale@flo zc.relation]$ echo 1 > foo
🌠 ~/tmp/zc.relation (master) 
[ale@flo zc.relation]$ git add foo 
🌠 ~/tmp/zc.relation (master) 
[ale@flo zc.relation]$ git commit -am "Test"
[master 076a091] Test
 1 file changed, 1 insertion(+)
 create mode 100644 foo
🌠 ~/tmp/zc.relation (master) 
[ale@flo zc.relation]$ git co -b test
Switched to a new branch 'test'
🌠 ~/tmp/zc.relation (test) 
[ale@flo zc.relation]$ git push
Enumerating objects: 4, done.
Counting objects: 100% (4/4), done.
Delta compression using up to 8 threads
Compressing objects: 100% (2/2), done.
Writing objects: 100% (3/3), 262 bytes | 262.00 KiB/s, done.
Total 3 (delta 1), reused 0 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (1/1), completed with 1 local object.
remote: fatal error in commit_refs
To github.com:zopefoundation/zc.relation.git
 ! [remote rejected] test -> test (failure)
error: failed to push some refs to 'github.com:zopefoundation/zc.relation.git'
```

Is it a permission problem? I am not even able to fork this package.

**Comment by ale-rt** [comment]
I do not know if it was something with git but now I am able to push again :)

**Review by ale-rt — APPROVED** [review]

**Comment by icemac** [comment]
@ale-rt GitHub had problems yesterday preventing from pushing, see https://www.githubstatus.com/incidents/52z0j6phhnjs.

**Comment by icemac** [comment]
Thank you for reviewing and fixing this PR.

File: src/zc/relation/catalog.py [hunk 1]
```diff
@@ -1177,13 +1180,13 @@ def canFind(self, query, maxDepth=None, filter=None,
...
-        try:
-            next(self.yieldRelationTokenChains(
+
+        _ = next(self.yieldRelationTokenChains(
                 *self._parse(
                     query, maxDepth, filter, targetQuery,
                     targetFilter, getQueries) +
-                (False,)))
-        except StopIteration:
+                (False,)), _marker)
```
Comment (zvezdan) [inline src/zc/relation/catalog.py:1188]: @ale-rt 
This line and lines 1185-1187 above need to be dedented by 4 spaces to the left.
Once we replaced `try-except` with the assignment this whole expression needs one indentation level less.

Comment (ale-rt) [reply src/zc/relation/catalog.py:1188]: Thanks @zvezdan!
Makes sense, but not really necessary.

I might come back to this, anyway!

Comment (zvezdan) [reply src/zc/relation/catalog.py:1188]: @ale-rt Of course. It's not necessary, but it's changing the code style of the package. A small nit. 🙂 
I just wanted to bring it to your attention.
I see that you're preparing other changes, so this small style fix could perhaps be included in the upcoming changes.
I'll resolve the conversation.
Thanks!

**Pull request merged**